Run github action against tag during release

glasser opened the pull request:
Previously this would pass the empty string which would default to main.

glasser committed:
Run github action against tag during release

Previously this would pass the empty string which would default to main.

jonathanrainer reviewed (approved):
LGTM

I was definitely mislead by the name of the option here, though I wouldn't worry, this stuff should be going in the bin when we consolidate the testing as I have a PR open to do

glasser committed:
Rename release version of job

If I understand correctly, before this PR the two jobs that run GitHub
actions were defined identically and magically merged by GitHub into a
single job definition. But now that they are slightly different, GH
treats them as different jobs, and adds `-1` and `-2` to their names. It
doesn't seem to mess up the `requires` links, but it is (a) tacky and
(b) messed up the GitHub required status on PRs. So differentiate the
names.
